### PR TITLE
Bring person detail together

### PR DESCRIPTION
**Pull request by abhiahirv1**
Creating a new subdoc called
”applicant_details”

and moving the following attributes under it.

first_name,

middle_name,

last_name,

date_of_birth,

nationality,

country_of_residence,

correspondence_address,

correspondence_address_is_same_as_registered_office_address.

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/models/dto/AcspDataDto.java [hunk 1]
```diff
@@ -5,27 +5,17 @@
 import uk.gov.companieshouse.acsp.models.enums.TypeOfBusiness;
 import uk.gov.companieshouse.acsp.models.type.Address;
 
-import java.time.LocalDate;
-
 public class AcspDataDto {
 
+    public ApplicantDetailsDto getApplicantDetails;
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/models/dto/AcspDataDto.java:10]: I dont think this is required

File: src/main/java/uk/gov/companieshouse/acsp/models/dto/ApplicantDetailsDto.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package uk.gov.companieshouse.acsp.models.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.companieshouse.acsp.models.type.Address;
+
+import java.time.LocalDate;
+
+public class ApplicantDetailsDto {
+
+    @JsonProperty("firstName")
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/models/dto/ApplicantDetailsDto.java:10]: Talk to @swalke-ch about the using the snake_case property names.

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java [hunk 3]
```diff
@@ -101,9 +101,9 @@ private void buildFilingStatus(FilingApi filing) {
   private void buildPresenter(Map<String, Object>data, AcspDataDto acspDataDto) {
     var presenter = new Presenter();
 
-    presenter.setFirstName(Optional.ofNullable(acspDataDto.getFirstName())
+    presenter.setFirstName(Optional.ofNullable(acspDataDto.getApplicantDetails().getFirstName())
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:104]: you will have to check if applicant details is null or not else its a potential NPE

Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:104]: I would say create a method like buildApplicantDetails where you build the details.

File: src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java [hunk 4]
```diff
@@ -103,26 +99,30 @@ void init() {
     private void setACSPDataDto() {
         acspDataDto = new AcspDataDto();
         acspDataDto.setId(ACSP_ID);
-        acspDataDto.setFirstName(FIRST_NAME);
-        acspDataDto.setLastName(LAST_NAME);
+
+        if (acspDataDto.getApplicantDetails() == null) {
```
Comment (sampankumar) [inline src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java:103]: We know its null at this point... no need to check for null... you can just create the new applicantdetails everytime

File: src/main/java/uk/gov/companieshouse/acsp/models/dto/AcspDataDto.java (hunk 1 above)
Comment (abhiahirv1) [reply src/main/java/uk/gov/companieshouse/acsp/models/dto/AcspDataDto.java]: Removed

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java (hunk 3 above)
Comment (abhiahirv1) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java]: Done

File: src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java (hunk 4 above)
Comment (abhiahirv1) [reply src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java]: I have removed. Thanks

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java [hunk 5]
```diff
@@ -98,20 +98,28 @@ private void buildFilingStatus(FilingApi filing) {
...
-            .map(String::toUpperCase).orElse(null));
-    presenter.setLastName(Optional.ofNullable(acspDataDto.getLastName())
-            .map(String::toUpperCase).orElse(null));
+    if (acspDataDto.getApplicantDetails() != null) {
+      presenter.setFirstName(Optional.ofNullable(acspDataDto.getApplicantDetails().getFirstName())
+              .map(String::toUpperCase).orElse(null));
+      presenter.setLastName(Optional.ofNullable(acspDataDto.getApplicantDetails().getLastName())
+              .map(String::toUpperCase).orElse(null));
+    } else {
+      presenter.setFirstName(null);
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:110]: Do you have to set these to null explicitly?

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java [hunk 6]
```diff
@@ -139,7 +147,7 @@ private void buildAcspData(Map<String, Object>data, AcspDataDto acspDataDto, Tra
       data.put("registered_office_address", buildBusinessAddress(acspDataDto));
     }
     if(acspDataDto.getBusinessAddress() != null &&
-            acspDataDto.getBusinessAddress().equals(acspDataDto.getCorrespondenceAddress())) {
+            acspDataDto.getBusinessAddress().equals(acspDataDto.getApplicantDetails().getCorrespondenceAddress())) {
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:150]: null check required here for applicantDetails

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java (hunk 5 above)
Comment (abhiahirv1) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java]: No

File: src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java (hunk 6 above)
Comment (abhiahirv1) [reply src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java:147]: added.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java [hunk 7]
```diff
@@ -7,11 +7,7 @@
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.test.util.ReflectionTestUtils;
-import uk.gov.companieshouse.acsp.models.dto.AcspDataDto;
-import uk.gov.companieshouse.acsp.models.dto.CompanyDto;
-import uk.gov.companieshouse.acsp.models.dto.NationalityDto;
-import uk.gov.companieshouse.acsp.models.dto.AMLSupervisoryBodiesDto;
-import uk.gov.companieshouse.acsp.models.dto.AcspDataSubmissionDto;
+import uk.gov.companieshouse.acsp.models.dto.*;
```
Comment (sampankumar) [inline src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java:10]: dont use * in imports

Comment (abhiahirv1) [reply src/test/java/uk/gov/companieshouse/acsp/service/FilingServiceTest.java]: Done

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by ttingle-ch — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

**Pull request merged**